amr-wind/boundary_conditions/wall_models/WallFunction.cpp

```diff
@@ -116,14 +128,103 @@ void VelWallFunc::wall_model(
+                            const amrex::Real v_dx =
+                                vold_arr(i, j, k_nearest, 1);
+
+                            varr(i, j, k - 1, 0) =
+                                tau.get_shear(uu, wspd, u_dx, v_dx, xc, 0) /
+                                mu * den(i, j, k);
+                            varr(i, j, k - 1, 1) =
+                                tau.get_shear(vv, wspd, u_dx, v_dx, xc, 1) /
+                                mu * den(i, j, k);
+                        } else {
+                            // Case where dx[0] > dx[2]
+                            const int k_lo = static_cast<int>(
+                                std::floor(dx[0] / dx[2] - 0.5));
+                            const int k_hi = k_lo + 1;
+
+                            // Ensure the interpolation points are within the
+                            // valid box
+                            if (bx.contains(amrex::IntVect(i, j, k_lo)) &&
+                                bx.contains(amrex::IntVect(i, j, k_hi))) {
+                                const amrex::Real z_lo = (k_lo + 0.5) * dx[2];
+                                const amrex::Real z_hi = (k_hi + 0.5) * dx[2];
+                                const amrex::Real z_interp = dx[0];
+
+                                // Check if the interpolation condition is
+                                // satisfied
+                                if (z_lo < z_interp && z_interp < z_hi) {
+                                    const amrex::Real weight =
+                                        (z_interp - z_lo) / (z_hi - z_lo);
+                                    const amrex::Real u_low =
+                                        vold_arr(i, j, k_lo, 0);
+                                    const amrex::Real u_up =
+                                        vold_arr(i, j, k_hi, 0);
+                                    const amrex::Real v_low =
+                                        vold_arr(i, j, k_lo, 1);
+                                    const amrex::Real v_up =
+                                        vold_arr(i, j, k_hi, 1);
+
+                                    const amrex::Real u_dx =
+                                        u_low + (u_up - u_low) * weight;
+                                    const amrex::Real v_dx =
+                                        v_low + (v_up - v_low) * weight;
+
+                                    varr(i, j, k - 1, 0) =
+                                        tau.get_shear(
+                                            uu, wspd, u_dx, v_dx, xc, 0) /
+                                        mu * den(i, j, k);
+                                    varr(i, j, k - 1, 1) =
+                                        tau.get_shear(
+                                            vv, wspd, u_dx, v_dx, xc, 1) /
+                                        mu * den(i, j, k);
+                                } else {
```

Comment:
I'm curious, when do we expect this branch to occur?

Reply:
Looking through the changes, did this get resolved?

Reply:
I believe it did get solve. Or perhaps I am not understanding the issue

Reply:
I'm trying to figure out when this branch will occur with all the new interpolation changes.

Reply:
I'll take a look at this tomorrow and remove the branch if it's no longer needed.

Reply:
@marchdf I Updated the checks here, should be good to go here

Reply:
well done. thank you!